content/tutorials/react-native-presence-publish-subscribe.textile

```diff
@@ -1,155 +1,152 @@
-# ably: This library is the official @Ably@ library for Node js
-# ably-react-native: This library is the official @Ably@ library for React Native
+# Express: This is a Node.js web framework which we’ll use to create our API.
+# Body-parser: This library is used by Express to parse body requests.
+# Ably: This library is the official @Ably@ library for Node js
+# Ably-react-native: This library is the official @Ably@ library for React Native
 
+"See this step in Github":https://github.com/samuelayo/tutorials/commit/b321b24c694cd9d3d67b9724132eb46da7fa1821
 
-h2. Creating Our API
+h2.
+  default: Step 3 - Creating Our API
 
 First, Create a new file called @server.js@, which serves as your API in our root folder and paste the following:
 
 ```[javascript]
 //require express
 var express = require('express')
-//define app as in instance of express
+    //define app as in instance of express
 var app = express()
-//require bosy-parser
+    //require bosy-parser
 var bodyParser = require('body-parser')
 
 //use bodyparser as a middle ware
 app.use(bodyParser.json())
+
 //ably realtime
 const ably = require('ably').Realtime;
-// instantiate an instance of ably passing in our API key
-const myably = new ably('XXX_API_KEY')
-//set the channel you will like to publish to 
-const channel = myably.channels.get('cog-tie');
+const ablyRealtime = new ably('XXX_API_KEY')
+const channel = ablyRealtime.channels.get('ably-chat');
+
+
+
 //set cors middleware
-app.use(function(req, res, next) {
-res.header("Access-Control-Allow-Origin", "*");
-res.header("Access-Control-Allow-Headers", "Origin, X-Requested-With, Content-Type, Accept");
-next();
+app.use((req, res, next)=> {
+    res.header("Access-Control-Allow-Origin", "*");
+    res.header("Access-Control-Allow-Headers", "Origin, X-Requested-With, Content-Type, Accept");
+    next();
 });
-//handle route postfunction
-app.post('/', function(req, res) {
-var message = {
-    user: req.body.name,
-    msg: req.body.msg,
-    action: req.body.action
-}
-//Publish your message
-channel.publish('message', message, function(err) {
-    if (err) {
-        console.log('publish failed with error ' + err);
-    } else {
-        console.log('publish succeeded ' + message.msg);
+//handle sending of message via ably
+app.post('/send_message', (req, res)=> {
+    var data = {
+        user: req.body.name,
+        msg: req.body.msg,
+        action: req.body.action
     }
-})
-res.send({ status: 'okay', message: message })
+    channel.publish('data', data, (err)=> {
+        if (err) {
+            console.log('publish failed with error ' + err);
+        } else {
+            console.log('publish succeeded ' + data.msg);
+        }
+    })
+    res.send({ status: 'okay', data: data });
 
-})
+});
 
-//listen on port and serve the app
-app.listen(3000, function() {
-console.log('Example app listening on port 3000!')
-})
+    //listen on port and serve the app
```

Comment:
Funky indentation again?

Reply:
Sorry about this, fixed it now with the recent push.